gprofiler/application_separators.py

```diff
@@ -0,0 +1,238 @@
+        for arg in process.cmdline():
+            if ":" in arg:
+                if _IP_PORT_RE.match(arg):
+                    continue
+
+                return append_python_module_to_proc_wd(process, arg)
+
+        logger.warning(f"GunicornApplicationSeparator: matched against process {process} but couldn't find WSGI module")
+        return None
+
+
+class UwsgiApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "uwsgi"
+
+    def is_supported(self, process: Process) -> bool:
+        return "uwsgi" in process.cmdline()[0]
+
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        wsgi = get_cli_arg_by_name(process.cmdline(), "-w")
+        if wsgi is not None:
+            return append_python_module_to_proc_wd(process, wsgi)
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return append_python_module_to_proc_wd(process, wsgi)
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
+        return None
+
+
+class CeleryApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "celery"
+
+    def is_supported(self, process: Process) -> bool:
+        if get_cli_arg_by_index(process.cmdline(), 0) == "celery":
+            return True
+
+        return len(process.cmdline()) >= 3 and ["-m", "celery"] == process.cmdline()[1:3]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        app_name = get_cli_arg_by_name(process.cmdline(), "-A") or get_cli_arg_by_name(
+            process.cmdline(), "--app", check_for_equals_arg=True
+        )
+        if app_name is None:
+            logger.warning("Couldn't find positional argument -A or --app for application indication")
+            return None
+
+        return app_name
+
+
+class PySparkApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "PySpark"
+
+    def is_supported(self, process: Process) -> bool:
+        # We're looking for pythonXX -m pyspark.daemon
+        return (
+            len(process.cmdline()) >= 3
+            and "python" in process.cmdline()
```

Comment:
In the following line you check `["-m", "pyspark.daemon"] == process.cmdline()[1:]` which suggests you meant
```suggestion
            len(process.cmdline()) == 3
            and "python" in process.cmdline()[0]
```

Reply:
Done, resolving.